src/ui/osx/TogglDesktop/Features/Timer/TimerViewController.swift

```diff
@@ -0,0 +1,533 @@
+                self.closeTagsAutoComplete()
+            }
+        }
+    }
+
+    private func closeProjectAutoComplete() {
+        projectAutoCompleteWindow.cancel()
+        projectButton.controlState = .normal
+        projectAutoCompleteView.clean()
+    }
+
+    private func closeTagsAutoComplete() {
+        tagsAutoCompleteWindow.cancel()
+        tagsButton.controlState = .normal
+        tagsAutoCompleteView.clean()
+    }
+
+    // MARK: - Other
+
+    @objc
+    private func clearProject() {
+        viewModel.clearProject()
+    }
+}
+
+// MARK: - NSTextFieldDelegate
+
+extension TimerViewController: NSTextFieldDelegate {
+    //
+}
+
+// MARK: - AutoCompleteViewDelegate
+
+extension TimerViewController: AutoCompleteViewDelegate {
+
+    func shouldClose() {
+        if projectAutoCompleteWindow.isVisible {
+            closeProjectAutoComplete()
+        } else if tagsAutoCompleteWindow.isVisible {
+            closeTagsAutoComplete()
+        }
+    }
+
+    func didTapOnCreateButton() {
+        if projectAutoCompleteWindow.isVisible {
+            showProjectCreationView()
+        } else if tagsAutoCompleteWindow.isVisible {
+            tagsAutocompleteDidTapOnCreateButton()
+        }
+    }
+
+    private func showProjectCreationView() {
+        projectCreationView.workspaceID = viewModel.workspaceID
+        projectCreationView.timeEntryGUID = viewModel.timeEntryGUID
+        projectCreationView.timeEntryIsBillable = viewModel.billableState == .on
+
+        updateProjectAutocompleteWindowContent(with: projectCreationView, height: projectCreationView.suitableHeight)
+        projectCreationView.setTitleAndFocus(projectAutoCompleteView.defaultTextField.stringValue)
+    }
+
+    private func updateProjectAutocompleteWindowContent(with view: NSView, height: CGFloat) {
+        projectAutoCompleteWindow.contentView = view
+
+        let fromPoint = NSPoint(x: projectButton.frame.minX, y: projectButton.frame.maxY)
+        let windowRect = autoCompleteWindowRect(fromPoint: fromPoint)
+        projectAutoCompleteWindow.setFrame(windowRect, display: false)
+        projectAutoCompleteWindow.setFrameTopLeftPoint(windowRect.origin)
+
+        projectAutoCompleteWindow.makeKey()
+    }
+
+    private func tagsAutocompleteDidTapOnCreateButton() {
+        viewModel.createNewTag(withName: tagsAutoCompleteView.defaultTextField.stringValue)
+        _ = tagsAutoCompleteView.defaultTextField.becomeFirstResponder()
+    }
+}
+
+// MARK: - ProjectCreationViewDelegate
+
+extension TimerViewController: ProjectCreationViewDelegate {
+    func projectCreationDidCancel() {
+        closeProjectAutoComplete()
+    }
+
+    func projectCreationDidAdd(with name: String, color: String, projectGUID: String) {
+        closeProjectAutoComplete()
+
+        // TODO: set project on UI
+        // problem is we don't have library method to get project by GUID and pass it to view model
```

Comment:
perhaps open an issue for this?

Reply:
opened an issue for this https://github.com/toggl-open-source/toggldesktop/issues/4334